aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,272 @@
+
+.. literalinclude:: routing/sample/RoutingSample/HelloRouter.cs
+  :linenos:
+  :language: c#
+
+``HelloRouter`` checks to see if ``RouteData`` includes a value for the key ``name``. If not, it immediately returns without handling the request. Otherwise, the request is handled (by writing out "Hi {name}!") and the ``RouteContext`` is updated to note that the request was handled. This prevents additional routes from handling the request.
+
+This route was configured to use an inline constraint (see below), signified by the ":alpha" in the route template for the name route value. This constraint limits which requests this route will handle, in this case to alphabetical values for ``name``. Thus, a request for "/hello/steve" will be handled, but a request to "/hello/123" will not (instead, the request will pass through to the "Hello World!" request delegate).
+
+Template Routes
+---------------
+The most common way to define routes is using ``TemplateRoute`` and route template strings. In a typical MVC app, you might use a default template route with a string like this one: 
+
+.. image:: /fundamentals/routing/_static/default-mvc-routetemplate.png
+
+Tokens within curly braces (``{ }``) define `route value` placeholders which will be bound if the route is matched. You can define more than one route value placeholder in a route segment, but they must be separated by a literal value. For example "{controller=Home}{action=Index}" would not be a valid route, since there is no literal value between ``{controller}`` and ``{action}``. These route value placeholders must have a name, and may have additional attributes specified.
+
+The following table demonstrates the available route value placeholder options and `route value constraints <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteOptions.cs>`_.
+
+.. list-table:: Route Template Values
+	:header-rows: 1
+
+	* - Route Template
+	  - Example Matching URL
+	  - Notes
+	* - hello
+	  - /hello
+	  - Will only match the single path '/hello'
+	* - {Page=Home}
+	  - /
+	  - Will match and set ``Page`` to ``Home``.
+	* - {Page=Home}
+	  - /Contact
+	  - Will match and set ``Page`` to ``Contact``
+	* - {controller}/{action}/{id?}
+	  - /Products/List
+	  - Will map to ``Products`` controller and ``List`` method; Since ``id`` was not supplied in the URL, it's ignored.
+	* - {controller}/{action}/{id?}
+	  - /Products/Details/123
+	  - Will map to ``Products`` controller and ``List`` method, with ``id`` set to ``123``.
+	* - {controller=Home}/{action=Index}/{id?}
+	  - /
+	  - Will map to ``Home`` controller and ``Index`` method; ``id`` is ignored.
+
+Adding a colon ``:`` after the name allows additional constraints to be set on a route value placeholder.
+
+.. list-table:: Route Constraints
+	:header-rows: 1
+
+	* - constraint
```

Comment:
I would also mention that using constraints for **validation** is a bit of a trap. It means that invalid input will result in a 404 instead of a 400 with a good error message. These constraints should be used to **disambiguate** between routes.